### PR TITLE
Update changelog for 0.35.0rc1

### PR DESCRIPTION
**Pull request by sklam**
as titled.

### CODE REVIEW

**Review by esc — APPROVED** [review]
LGMT!

File: CHANGE_LOG [hunk 1]
```diff
@@ -1,7 +1,36 @@
-v0.35.0dev
------------
+v0.35.0rc1
```
Comment (stuartarchibald) [inline CHANGE_LOG:1]: Want to put a stub in for the date so we don't forget to add it later?

File: CHANGE_LOG [hunk 2]
```diff
@@ -1,7 +1,36 @@
...
+The main feature in this release is a Numba specific LLVM pass for pruning
+reference-count operations. We plan to generalize this custom LLVM pass
+once it is proven stable so that it can be configured for other uses.
+In addition, this release contains an updated SVML patch that fixes an issue
+for AVX512.
+
+Pull requests:
+
+* PR #614: Fix wheel building.
+* PR #616: fix release date in changelog
```
Comment (stuartarchibald) [inline CHANGE_LOG:13]: Is this on mainline?

File: CHANGE_LOG [hunk 3]
```diff
@@ -1,7 +1,36 @@
...
------------
+v0.35.0rc1
+----------
 
-In development
+The main feature in this release is a Numba specific LLVM pass for pruning
+reference-count operations. We plan to generalize this custom LLVM pass
+once it is proven stable so that it can be configured for other uses.
+In addition, this release contains an updated SVML patch that fixes an issue
+for AVX512.
```
Comment (stuartarchibald) [inline CHANGE_LOG:8]: It also sorts out issues with fPIC and support for BSD.

**Review by stuartarchibald — COMMENTED** [review]
Few minor comments else looks good, thanks.

File: CHANGE_LOG (hunk 2 above)
Comment (sklam) [reply CHANGE_LOG:13]: good catch

File: CHANGE_LOG (hunk 3 above)
Comment (sklam) [reply CHANGE_LOG:8]: I'll add that

File: CHANGE_LOG (hunk 2 above)
Comment (sklam) [reply CHANGE_LOG:13]: i'll also need to update my script to check for the target branch

**Review by stuartarchibald — APPROVED** [review]
Thanks for the patch and fixes.

**Pull request merged**